drivers/dma/dma_silabs_siwx917.c

```diff
@@ -0,0 +1,634 @@
+	UDMA_Channel_Info siwx917_udma_channel_info; // UDMA channel info
+	dma_callback_t dma_callback;                 // User callback
+	void *cb_data;                               // User callback data
+};
+
+/*Function to validate and convert channel priority for sl layer usage*/
+static inline int siwx917_dma_priority(uint32_t priority)
+{
+	switch (priority) {
+	case 0:
+		return PRIORITY_LOW;
+	case 1:
+		return PRIORITY_HIGH;
+	default:
+		printf("Invalid DMA priority - must be low priority (0), high priority (1)");
+		return -1;
+	}
+	return 0;
+}
+
+/*Function to validate and convert channel transfer direction for sl layer
+ * usage*/
+static inline int siwx917_dma_transfer_direction(uint32_t dir)
+{
+	switch (dir) {
+	case 0:
+		return PERIPHERAL_REQUEST_DISABLE; // Memory to Memory transfer
+	case 1:
+		return PERIPHERAL_REQUEST_ENABLE; // Memory to Peripheral transfer
+	case 2:
+		return PERIPHERAL_REQUEST_ENABLE; // Peripheral to Memory transfer
+	default:
+		printf("Invalid Transfer direction - must be,\n \
+               memory-memory (0)\n \
+               memory-peripheral (1)\n \
+               peripheral-memory (2)\n");
+		return -1;
+	}
+	return 0;
+}
+
+/*Function to validate and convert channel data width for sl layer usage*/
+static inline int siwx917_dma_data_width(uint32_t data_width)
+{
+	switch (data_width) {
+	case 1:
+		return SL_TRANSFER_SIZE_8; // 8-bit width
+	case 2:
+		return SL_TRANSFER_SIZE_16; // 16-bit width
+	case 4:
+		return SL_TRANSFER_SIZE_32; // 32-bit width
+	default:
+		printf("Invalid Source/Destination data size - must be,\n \
+               8-bit (1)\n \
+               16-bit (2)\n \
+               32-bit (4)\n");
+		return -1;
+	}
+	return 0;
+}
+
+/*Function to validate and convert channel burst length for sl layer usage*/
+static inline int siwx917_dma_burst_length(uint32_t blen)
+{
+	switch (blen) {
+	case 1:
+		return VALID_BURST_LENGTH; // 8-bit burst
+	default:
+		printf("Invalid Source/Destination burst length - must be,\n \
+               8-bit Increment(1)\n");
+		return -1;
+	}
+	return 0;
+}
+
+/*Function to validate and convert channel addr increment value for sl layer
+ * usage*/
+static inline int siwx917_dma_addr_adjustment(uint32_t adjustment)
+{
+	switch (adjustment) {
+	case 0:
+		break; // Addr Increment
+	case 2:
+		return UDMA_ADDR_INC_NONE; // No Address increment
+	default:
+		printf("Invalid Source/Destination Addr adjustment - must be,\n \
+               Increment (0)\n \
+               No change (2)\n");
+		return -1;
+	}
+	return 0;
+}
+
+/*Function to configure UDMA channel for transfer*/
+static int dma_siwx917_configure(const struct device *dev, uint32_t channel,
+				 struct dma_config *config)
+{
+
+	// Fetch DMA config and data structures from device.
+	const struct dma_siwx917_config *cfg = dev->config;
+	struct dma_siwx917_data *data = dev->data;
+
+	uint32_t dma_instance;
+	sl_status_t status = SL_STATUS_OK;
+	UDMA_RESOURCES UDMA_Resources;
```

Comment:
I suggest to initialize this struct during declaration. Otherwise if someone add a member in `UDMA_RESOURCES`, it won't be initialized.

In contrary, it does not really make sense to initialize `status` since you are going to overwrite it anyways.

Reply:
Done